pkg/build/push.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"bytes"
+
+	log "github.com/Sirupsen/logrus"
+	dockerlib "github.com/fsouza/go-dockerclient"
+	"github.com/novln/docker-parser"
+	"github.com/pkg/errors"
+)
+
+// Push will provide methods for interaction with API regarding pushing images
+type Push struct {
```

Comment:
Why is there a separate struct for Push? Does it hold any special information or data?
Would it make sense to have `PushImage` function on `Build` struct?

Reply:
This code was taken from https://github.com/kubernetes/kompose/blob/f42450b2e8a980725a5240e3c33641bca9b293c6/pkg/utils/docker/push.go#L28 with little changes, so did not really investigate why two structs were defined!

Maybe @cdrage can answer on this one!